packages/terra-pills/src/private/_Pill.jsx

```diff
@@ -0,0 +1,301 @@
+/* eslint-disable jsx-a11y/role-supports-aria-props */ // lp052179
+import React, {
+  useCallback, useLayoutEffect, useRef, useState,
+} from 'react';
+import {
+  KEY_SPACE,
+  KEY_RETURN,
+  KEY_DELETE,
+  KEY_BACK_SPACE,
+} from 'keycode-js';
+import { injectIntl } from 'react-intl';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import classNamesBind from 'classnames/bind';
+import Popup from 'terra-popup';
+import ResizeObserver from 'resize-observer-polyfill';
+import ThemeContext from 'terra-theme-context';
+import styles from './Pill.module.scss';
+
+const cx = classNamesBind.bind(styles);
+
+const propTypes = {
+  /**
+    * The html 'id' attribute for the pill, needed for proper keyboard interactions. (Required & must be unique)
+    */
+  id: PropTypes.string.isRequired,
+  /**
+   * The label text for the pill. (Required)
+   */
+  label: PropTypes.string.isRequired,
+  /**
+   * Secondary label text for the pill that displays a filter category
+   */
+  labelCategory: PropTypes.string,
+  /**
+   * Object returned along with the 'pillKey' in the `onRemove`
+   */
+  metaData: PropTypes.object,
+  /**
+   * To identify which pill is being removed.
+   */
+  pillKey: PropTypes.string,
+  /**
+   * @private
+   * The intl object to be injected for translations.
+   */
+  intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
+  /**
+   * A callback function to execute when the pill is removed. Returns pillKey, metadata.
+   */
+  onRemove: PropTypes.func,
+  /**
+   * @private
+   * Tooltip to display if the pill label does not have enough space to display and will show as truncated,
+   * to be used in addition with a popup.
+   */
+  title: PropTypes.string,
+};
+
+const defaultProps = {
+  onRemove: undefined,
+  pillKey: undefined,
+  title: undefined,
+};
+
+const Pill = (props) => {
+  const {
+    label,
+    labelCategory,
+    id,
+    intl,
+    metaData,
+    onRemove,
+    pillKey,
+    title,
+    onSelect,
+    ...customProps
+  } = props;
+
+  const theme = React.useContext(ThemeContext);
+  const pillRef = useRef();
+  const [isTruncated, setIsTruncated] = useState(false);
+  const [open, setPopupOpen] = useState(false);
```

Comment:
The name of the state should be more consistent so we should either update `open` to `popupOpen` or update `setPopupOpen` to `setOpen`.